course/templatetags/course.py

```diff
@@ -124,6 +125,15 @@ def tags(profile, instance):
     tags = UserTagging.objects.get_all(profile, instance)
     return mark_safe(' '.join(tag.html_label for tag in tags))
 
+@register.simple_tag
+def api_token(profile, instance):
+    payload = {}
+    if profile:
+        payload['uid'] = profile.id
```

Comment:
There is no mitch match. You add `user.userprofile.id` to `uid` and compare that later to `user.id`, which are different.

Reply:
Would be better to store the `user.id`, which can be get from profile with `profile.user.id`.